executor/distsql.go

```diff
@@ -819,7 +833,20 @@ func (e *XSelectTableExec) doRequest() error {
 	selReq.Where = e.where
 	selReq.TableInfo = &tipb.TableInfo{
 		TableId: e.tableInfo.ID,
-		Columns: distsql.ColumnsToProto(e.Columns, e.tableInfo.PKIsHandle),
+	}
+	selReq.TableInfo.Columns = distsql.ColumnsToProto(e.Columns, e.tableInfo.PKIsHandle)
+	for i, c := range e.Columns {
+		if c.State == model.StatePublic {
+			continue
+		}
+		d, _, err := table.GetColDefaultValue(e.ctx, c)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		selReq.TableInfo.Columns[i].DefaultVal, err = tablecodec.EncodeValue(d)
```

Comment:
I'm not familiar with this, but when the default value of a column is changed twice, which value will be used?